lib/remap-functions/src/parse_url.rs

```diff
@@ -58,34 +56,32 @@ impl Expression for ParseUrlFn {
     }
 }
 
-impl From<Url> for event::Value {
-    fn from(url: Url) -> Self {
-        let mut map = BTreeMap::<&str, event::Value>::new();
-
-        map.insert("scheme", url.scheme().to_owned().into());
-        map.insert("username", url.username().to_owned().into());
-        map.insert(
-            "password",
-            url.password()
-                .map(ToOwned::to_owned)
-                .unwrap_or_default()
-                .into(),
-        );
-        map.insert("path", url.path().to_owned().into());
-        map.insert("host", url.host_str().map(ToOwned::to_owned).into());
-        map.insert("port", url.port().map(|v| v as isize).into());
-        map.insert("fragment", url.fragment().map(ToOwned::to_owned).into());
-        map.insert(
-            "query",
-            url.query_pairs()
-                .into_owned()
-                .map(|(k, v)| (k, v.into()))
-                .collect::<BTreeMap<String, event::Value>>()
-                .into(),
-        );
-
-        event::Value::from_iter(map.into_iter().map(|(k, v)| (k.to_owned(), v)))
-    }
+fn url_to_value(url: Url) -> Value {
+    let mut map = BTreeMap::<&str, Value>::new();
+
+    map.insert("scheme", url.scheme().to_owned().into());
+    map.insert("username", url.username().to_owned().into());
+    map.insert(
+        "password",
+        url.password()
+            .map(ToOwned::to_owned)
+            .unwrap_or_default()
+            .into(),
+    );
+    map.insert("path", url.path().to_owned().into());
+    map.insert("host", url.host_str().map(ToOwned::to_owned).into());
+    map.insert("port", url.port().map(|v| v as i64).into());
+    map.insert("fragment", url.fragment().map(ToOwned::to_owned).into());
+    map.insert(
+        "query",
+        url.query_pairs()
+            .into_owned()
+            .map(|(k, v)| (k, v.into()))
+            .collect::<BTreeMap<String, Value>>()
+            .into(),
+    );
+
+    Value::from_iter(map.into_iter().map(|(k, v)| (k.to_owned(), v)))
```

Comment:
This `map` has a fixed number of elements, and the tree form is not actually used other than to iterate over. I think this could be done as an array (`[(&str, Value)]`) and avoid the extra tree allocation work for nothing.

Yes, this is just a code move, didn't realize this before.

Reply:
Yeah, this can definitely be improved. We still have a "performance pass" to do over all of our functions, but I'll leave that for another PR.

I created an issue to track these at https://github.com/timberio/vector/issues/5573.